### PR TITLE
Optimize/download

### PR DESCRIPTION
**Pull request by lpoli**
This PR contains the following changes:
1. Remove redundancies
2. Make authticket mandatory for clients other than allocation owner and collaborator
3. Change tests for authticket being mandatory
4. Make share-info checking mandatory for all authtickets(public/private)

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | modify/download-test  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | optimize/download  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | modify/download-test  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | hotfix/staging-stabilization  |
| blobber  | optimize/download  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | modify/download-test  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | fix/public-share-issue  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | optimize/download  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: code/go/0chain.net/blobbercore/handler/object_operation_handler.go [hunk 1]
```diff
@@ -391,61 +358,38 @@ func (fsh *StorageHandler) DownloadFile(ctx context.Context, r *http.Request) (r
...
 		}
+
 		if err := encscheme.InitForDecryption("filetype:audio", fileref.EncryptedKey); err != nil {
 			return nil, err
 		}
 
 		totalSize := len(respData)
 		result := []byte{}
 		for i := 0; i < totalSize; i += int(fileref.ChunkSize) {
-			encMsg := &zencryption.EncryptedMessage{}
```
Comment (dabasov) [inline code/go/0chain.net/blobbercore/handler/object_operation_handler.go:443]: it is the same as new()

**Review by dabasov — APPROVED** [review]
lgtm

**Review by cnlangzi — DISMISSED** [review]
please get system-test passed first. thanks.

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | modify/download-test  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | fix/public-share-issue  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | optimize/download  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**